documentation/src/main/asciidoc/topics/con_clients_marshalled_objects.adoc

```diff
@@ -0,0 +1,25 @@
+[id='clients-marshalled-objects_{context}']
+= Clients and marshalled objects
+When you configure your client to use a marshaller, you must configure cache to use that marshaller's encoding.
```

Comment:
The more I think about this topic, the more I think making a parallel between marshalled objects in Hot Rod and REST clients doesn't makes sense.

1. Marshalling seems very Java-specific, and the REST endpoint is intended mostly for non-Java clients.
2. The server can transcode only a limited subset of application/json to application/x-protostream. Even assuming a non-Java client has a object-to-JSON marshaller, generating the correct JSON without ProtoStream is going to be tricky.

Note for @anistor: I don't think it's enough for the javadoc of `ProtobufUtil.toCanonicalJSON` to link to https://developers.google.com/protocol-buffers/docs/proto3#json, because there significant differences between the protobuf canonical representation and the protostream canonical representation. E.g. according to that page `Any` should be represented with a `@type` attribute embedded in the object, and none of the objects should have a `_type` attribute.

Reply:
@danberindei I think that makes a lot of sense. Will update.

Reply:
@danberindei Is "Any" part of protobuf2? We're not on 3 yet. Regarding the ```_type```, it's a Infinispan only extension. The canonical mapping describe the mapping from ```Message``` to ```Json object```, but when a cache can host several messages, there is an ambiguity that cannot be solved

Reply:
> 1. Marshalling seems very Java-specific, and the REST endpoint is intended mostly for non-Java clients.

You can use the REST API from Java, and then send marshalled objects with Content-Type ```application/x-java-serialized``` or read marshalled objects from an ```application/x-java-object``` cache

Reply:
I've reworked this section a little. I think maybe having a separate `REST clients` section feels wrong after what Dan commented so I put the emphasis more on ProtoStream to JSON conversion.

@gustavonalle I think we should perhaps leave that detail out if it's not a common case. I'm not sure we'd want to suggest that users should even do that by documenting it. Newer users might also find that confusing.